research/03-literate-prog.tex

```diff
@@ -1,46 +1,58 @@
 \section{Literate Programming}
 \label{sec:literate-programming}
+Just as with REPLs, the concept of literate programming is implemented in
+various forms under various names. Therefore this section starts with an
+explanation of what literate programming is exactly.
 
-Literate programming was first introduced in 1984 for the Pascal programming
-language~\cite{knuth1984}. Even though the idea was conceived over 30 years
-ago, implementations are not very common. In recent years the idea seems to
-gain popularity again. Examples include the IPython project~\cite{ipython2007},
-published in 2007, as well as Swift Playgrounds~\cite{swift-playgrounds},
-introduced in 2014.
-
-Just as with REPLs, the concept of literate programming is known by various
-names in slightly different forms. This section starts with an explanation of
-what literate programming is exactly.
-
-According to Donald Knuth, better documentation of programs is essential to make
-further progress in the state of the art of programming.
-To achieve this he proposes to write programs not with the intention to
-explain the computer what to do, but with the intention to explain to humans
-what the programmer wants a computer to do~\cite{knuth1984},
-by mixing documentation and code in a single source file.
-This idea of literate programming was realized in its original form as the
-``WEB'' language developed during Knuth's research at Stanford University.
+Literate programming, as defined in~\cite{knuth1984}, introduces the ability to
+annotate source code with natural language. According to Donald Knuth, better
+documentation of programs is essential to make further progress in the state of
+the art of programming.  To achieve this he proposes to write programs not with
+the intention to explain the computer what to do, but with the intention to
+explain to humans what the programmer wants a computer to do~\cite{knuth1984},
+by mixing documentation and source code in a single file. This idea of
+literate programming was realized in its original form as the ``WEB'' language
+developed during Knuth's research at Stanford University.
+
+Even though the idea was conceived over 30 years ago, implementations are not
+very common. However, in recent years the idea seems to gain popularity again.
+A very recent implementation of literate programming is Apple's Swift
+playgrounds~\cite{swift-playgrounds}. Swift playgrounds are interactive
+documents in which code is executed and run as it is typed, in contrast to the
+non-interactive style of the first ``WEB'' language.
```

Comment:
The *first* WEB language? Are there more than one? How about changing this to "of the first literate programming language"?